### ISSUE TITLE
Unable to use getter method in [textField] for dropdownlists

### ISSUE THREAD

**Issue by Rytrex** [telerik/kendo-angular#1682]
## I'm submitting a...

* Bug report 

## Current behavior

For dropdownlists, when `[data]` is an object array and `[textField]` is a getter method, it displays `[object Object]`.

## Expected behavior

I expect the dropdownlist component to display the returned value of the getter method. It did so in version 2.1.0 (currently using 3.0.2)

## Minimal reproduction of the problem with instructions

https://next.plnkr.co/edit/MhmDQ16VetqSEgnh
Select any value from the dropdown list; the component displays `[object Object]` instead of the string returned from the getter method.

## Environment

Browser:

- Chrome (desktop) version 67 

System:
- TypeScript version: 1.0.3.0
- Node version: 8.9.3, NPM 5.5.1
- Platform: Windows 

It seems like the popup component also does not display the correct text in the plunkr, however it does display the correct text in the application I'm working on (both are on version 2.4.1). Not sure if there is a related issue there.

**Comment by danielkaradachki** [comment]
You should set both the text and value fields. The `displayText` is not shown in the example in the popup because the objects are not instances of `TShirt`:
https://next.plnkr.co/edit/T6hTMlSpvq6NpPOj

**Comment by Rytrex** [comment]
Sorry about that, it seems like the issue is caused when using a getter method for `[textField]` and using `[valuePrimitive]="true"`. Please see https://next.plnkr.co/edit/SFsISUDys7ERVwD1.

**Comment by valchev** [comment]
Hi @Rytrex, I will label this as a bug and provide a fix as soon as possible.

**Comment by akloeber** [comment]
Right now I'm injecting a patched implementation of `hasOwnProperty` as a workaround:

```ts
  if (!type.prototype.hasOwnProperty('hasOwnProperty')) {
    type.prototype.hasOwnProperty = function hasOwnProperty(p) {
      if (Object.prototype.hasOwnProperty.call(this, p)) {
        return true;
      }

      let o = this;
      while (o = Object.getPrototypeOf(o)) {
        if (o === Object.prototype) {
          return false;
        }
        const pd = Object.getOwnPropertyDescriptor(o, p);
        if (pd && pd.get) {
          return true;
        }
      }
    };
  }

```

**Comment by elca-lan** [comment]
The same problem exists in the `ComboBoxComponent.prop` implementation. When using a getter property as the textField with `[valuePrimitive]="true"`, the focusing of items in the combobox popup stops working because `ComboBoxComponent.findIndexPredicate` will always compare the search bar value with the string "[object object]" which obviously doesn't do anything useful.

**Comment by dimitar-pechev** [comment]
The fix for both the **DropDownList** and the **ComboBox** is available in the latest [dev version](https://www.telerik.com/kendo-angular-ui/components/installation/development-builds/) of the package (`@progress/kendo-angular-dropdowns@3.5.0-dev.201901311002`). Here's [an example](https://stackblitz.com/edit/angular-h9przl) with the corrected behavior.

**Comment by akloeber** [comment]
Development build `@progress/kendo-angular-dropdowns@3.5.0-dev.201901311002` fixes the issue described in #1750.

**Comment by dimitar-pechev** [comment]
Shipped with `@progress/kendo-angular-dropdowns@3.5.0`.
